src/main/java/org/elasticsearch/search/aggregations/reducers/BucketHelpers.java

```diff
@@ -0,0 +1,187 @@
+         */
+        public String getName() {
+            return parseField.getPreferredName();
+        }
+    }
+
+    /**
+     * Given a path and a set of buckets, this method will return the value inside the agg at
+     * that path.  This is used to extract values for use by reducers (e.g. a derivative might need
+     * the price for each bucket).  If the bucket is empty, the configured GapPolicy is invoked to
+     * resolve the missing bucket
+     *
+     * @param histo      A series of agg buckets in the form of a histogram
+     * @param bucket     A specific bucket that a value needs to be extracted from.  This bucket should be present
+     *                   in the <code>histo</code> parameter
+     * @param aggPath    The path to a particular value that needs to be extracted.  This path should point to a metric
+     *                   inside the <code>bucket</code>
+     * @param gapPolicy  The gap policy to apply if empty buckets are found
+     * @return           The value extracted from <code>bucket</code> found at <code>aggPath</code>
+     */
+    public static Double resolveBucketValue(InternalMultiBucketAggregation<?, ? extends InternalMultiBucketAggregation.Bucket> agg,
+            InternalMultiBucketAggregation.Bucket bucket, String aggPath, GapPolicy gapPolicy) {
+        List<String> aggPathsList = AggregationPath.parse(aggPath).getPathElementsAsStringList();
+        return resolveBucketValue(agg, bucket, aggPathsList, gapPolicy);
+    }
+
+    public static Double resolveBucketValue(InternalMultiBucketAggregation<?, ? extends InternalMultiBucketAggregation.Bucket> agg,
+            InternalMultiBucketAggregation.Bucket bucket, List<String> aggPathAsList, GapPolicy gapPolicy) {
+        try {
+            Object propertyValue = bucket.getProperty(agg.getName(), aggPathAsList);
+            if (propertyValue == null) {
+                throw new AggregationExecutionException(DerivativeParser.BUCKETS_PATH.getPreferredName()
+                        + " must reference either a number value or a single value numeric metric aggregation");
+            } else {
+                double value;
+                if (propertyValue instanceof Number) {
+                    value = ((Number) propertyValue).doubleValue();
+                } else if (propertyValue instanceof InternalNumericMetricsAggregation.SingleValue) {
+                    value = ((InternalNumericMetricsAggregation.SingleValue) propertyValue).value();
+                } else {
+                    throw new AggregationExecutionException(DerivativeParser.BUCKETS_PATH.getPreferredName()
+                            + " must reference either a number value or a single value numeric metric aggregation");
+                }
+                // doc count never has missing values so gap policy doesn't apply here
+                boolean isDocCountProperty = aggPathAsList.size() == 1 && "_count".equals(aggPathAsList.get(0));
```

Comment:
Not sure it matters? If the whole path is `_count` then it will work as is, and you can't get deeper buckets if there is a doc count of 0 in the top level anyway?